### PR TITLE
Python export of MWRunFiles, DataSelector and WorkspaceSelector widgets

### LINKED ISSUE #28673: Python export of MWRunFiles, DataSelector and WorkspaceSelector widgets
These three widgets should be accessible in python code (we will actually need them in #25885)

* `MWRunFiles`: already accessible through `mantidqt.widgets.filefinder`. The renaming must be propagated in the C++ code.
* `DataSelector` and `WorkspaceSelector`: add the export, keeping the original name.

### PR DESCRIPTION
**Pull request by gui-co**
**Description of work.**

This PR provides a python export of `MWRunFiles`, `DataSelector` and `WorkspaceSelector` widgets in `mantidqt.widgets`.

The `MWRunFiles` widget, which was renamed `FileFinder` in python was renamed again (both in ptyhon and c++) `FileFinderWidget`. This will avoid confusion with `FileFinder` of `mantid.api`.

**To test:**

```python
from mantidqt.widgets.filefinderwidget import FileFinderWidget
from mantidqt.widgets.dataselector import DataSelector
from mantidqt.widgets.workspaceselector import WorkspaceSelector

w1 = FileFinderWidget()
w1.show()
w2 = DataSelector()
w2.show()
w3 = WorkspaceSelector()
w3.show()
```

Fixes #28673 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by gvardany** [comment]
@gui-co I'm not sure renaming the python export of `FileFinder` is safe; there might be user guis using that. I think for that you should rename only the C++ one. There is no big problem having FileFinder also in api, it's a different namespace.

**Comment by gui-co** [comment]
Following the comment, I changed back the python name of `MWRunFiles`:
`FileFinder` in `mantidqt.widgets.filefinder`


**To test:**
```python
from mantidqt.widgets.filefinder import FileFinder
from mantidqt.widgets.dataselector import DataSelector
from mantidqt.widgets.workspaceselector import WorkspaceSelector

w1 = FileFinder()
w1.show()
w2 = DataSelector()
w2.show()
w3 = WorkspaceSelector()
w3.show()
```

File: qt/python/mantidqt/widgets/filefinder/__init__.py [hunk 1]
```diff
@@ -9,4 +9,7 @@
 #
 from mantidqt.utils.qt import import_qt
 
-FileFinder = import_qt('..._common', 'mantidqt.widgets.filefinder', 'MWRunFiles')
+FileFinder = import_qt(
```
Comment (martyngigg) [inline qt/python/mantidqt/widgets/filefinder/__init__.py:12]: I like the idea of renaming this to be `FileFinderWidget` but as you have said we need to be careful of the API change. Can we deprecate the old with:

```python
FileFinderWidget = ...

def FileFinder(*args, **kwargs):
    from warnings import warn
    warn("The name FileFinder is deprecated, please use FileFinderWidget instead.")
    return FileFinderWidget(*args, **kwargs)
```
?

**Review by martyngigg — CHANGES_REQUESTED** [review]
This is a nice addition to the tools available in Python and it's good to have got rid of the confusing `MWRunFiles` name.
Just one question about the `FileFinderWidget` name.

**Comment by gui-co** [comment]
With the deprecated `FileFinder`

**To test:**
```python
from mantidqt.widgets.filefinder import FileFinder
from mantidqt.widgets.filefinderwidget import FileFinderWidget
from mantidqt.widgets.dataselector import DataSelector
from mantidqt.widgets.workspaceselector import WorkspaceSelector

w1 = FileFinder()
w1.show()
w2 = FileFinderWidget()
w2.show()
w3 = DataSelector()
w3.show()
w4 = WorkspaceSelector()
w4.show()
```

File: qt/python/mantidqt/widgets/filefinder/__init__.py (hunk 1 above)
Comment (gui-co) [reply qt/python/mantidqt/widgets/filefinder/__init__.py]: I put back the old `FileFinder` module. It returns a `FileFinderWidget` and log a warning.

**Review by martyngigg — DISMISSED** [review]
Thanks for making those changes. This looks good now.

**Comment by gui-co** [comment]
Thanks for the review.

File: qt/widgets/common/CMakeLists.txt [hunk 2]
```diff
@@ -55,7 +55,7 @@ set(
   src/Message.cpp
   src/MessageDisplay.cpp
   src/MultifitSetupDialog.cpp
-  src/MWRunFiles.cpp
+  src/FileFinderWidget.cpp
```
Comment (gvardany) [inline qt/widgets/common/CMakeLists.txt:58]: Can you sort these lists alphabetically?

Comment (gui-co) [reply qt/widgets/common/CMakeLists.txt]: Done

**Comment by gvardany** [comment]
Can you reapprove?

**Review by martyngigg — APPROVED** [review]

**Pull request merged**